### ISSUE TITLE
Suddenly stop working

### ISSUE THREAD

**Issue by kernal69** [marthoc/homeseer#68]
Hi

This is a great plugin when it works.
But it has a problem.
Suddenly it just stop working,
It happens mostly when i remove a device in Homeseer or make som other reconfiguration on Home Assistant.
If i restart Home Assistant everything works again.

A change in a home controlling system is not unusual.
So this is a problem.
I can not trust this plugin.

Could this be solved ?
If you need some logs or something just tell me what you need.
I really want this to work.

**Comment by marthoc** [comment]
Thanks for the report. You will need to provide logs (follow the instructions in the README to enable detailed logging) of when the problem occurs so that I can see what is reported.

**Comment by waynehead99** [comment]
@kernal69 I notice that this happens whenever you make a change to the homeseer integration... you need to restart Core to re-establish the connection. I haven't seen a retry mechanism  when the connection is lost.

**Comment by marthoc** [comment]
There is definitely a retry mechanism, the question is what changes are you making that leads to this state  - it may be the case that after certain things are updated the integration is not reaching the point where it is establishing a connection again. Debug logs would be helpful here.

**Comment by waynehead99** [comment]
All you need to do is restart HomeSeer and everything stops working. HA will not connect back without restarting the core. Reloading the integration itself is hit/miss on the reconnecting.
This tripped me up for a while until I realized what was happening.

**Comment by kernal69** [comment]
Hi
I can not send you logs on this.
Because of this problem i could not use it and i  have uninstalled it.
I use MQTT insted.

Maybe if it works in future i look into again.

**Comment by marthoc** [comment]
@waynehead99 Is it possible for you to enable debug logging per the instructions in the README and then get it to a non-working state? If you can paste the logs here I can hopefully have some evidence of why it is not reconnecting. See #70 which I believe is the same issue. I am happy to fix, but I need to see what's going wrong.

**Comment by TheBartoMan** [comment]
I agree with this. I'm constantly finding Home Assistant has stopped all communication with Homeseer.

A simple way for anyone to regenerate this is to restart Homeseer and you will find Home Assistant is no longer communicating with it .

I then do a restart of Home Assistant and it all comes back.

I wish Home Assistant would constantly check and reconnect.
